open_city_profile/decorators.py

```diff
@@ -56,6 +56,10 @@ def permission_checker(context):
     return permission_checker
 
 
+"""Decorator for checking that the user is logged in"""
```

Comment:
It's not possible to add a docstring to a variable. Even if it was, it would have to come on the next line, not previous.

This and the next docstring should therefore be regular comments.

Reply:
Ah, silly me. Making the string as documentation for the decorator could still be useful. That can be achieved at least like this:

```python
login_required = _build_decorator(_require_authenticated)
login_required.__doc__ = "Decorator for checking that the user is logged in"
```

Does this look ok, or is there a better way?

Reply:
Yeah, that looks ok.

Reply:
I did a bigger change for this. I figured that in order for the decorators to be "good citizens", in addition to `__doc__`, they would also need to have correct `__name__` and `__qualname__`. This would have led to:

```python
login_required = _build_decorator(_require_authenticated)
login_required.__name__ = "login_required"
login_required.__qualname__ = "login_required"
login_required.__doc__ = "Decorator for checking that the user is logged in"
```

That's not that simple any more and the function name is repeated six times.

So these new decorators aren't built by assigning to a variable any more, but instead they are real functions introduced with `def`. There is a helper decorator that they use. A docstring within the function works as expected.